### PR TITLE
[Feature] Add SRFacicalLandmarkDataset.

### PR DESCRIPTION
**Pull request by Yshuo-Li**

### CODE REVIEW

File: mmedit/datasets/sr_landmark_dataset.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from .base_sr_dataset import BaseSRDataset
+from .registry import DATASETS
+
+
+@DATASETS.register_module()
+class SRLandmarkDataset(BaseSRDataset):
+    """General image and landmark dataset with an annotation file for image
+    restoration.
+
+    The dataset loads gt (Ground-Truth) image, shape of image, face box, and
```
Comment (ckkelvinchan) [inline mmedit/datasets/sr_landmark_dataset.py:14]: Is it for general purposes or for faces only?

File: mmedit/datasets/sr_landmark_dataset.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        self.data_infos = self.load_annotations()
+
+    def load_annotations(self):
+        """Load annoations for SR dataset.
+
+        Annotation file is a `npy` file, which contains a list of dict.
+
+        It loads the GT image path and landmark from the annotation file.
+        Each dict in the annotation file contains the image names, image
+        shape (usually for gt), bbox and landmark separated by a white space.
```
Comment (ckkelvinchan) [inline mmedit/datasets/sr_landmark_dataset.py:52]: If it is a dict, why are they separated by a white space?

File: mmedit/datasets/sr_landmark_dataset.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        Returns:
+            dict: Returned dict for GT and landmark.
+                Contains: gt_path, bbox, shape, landmark.
+        """
+        data_infos = np.load(self.ann_file, allow_pickle=True)
+        for data_info in data_infos:
+            data_info['gt_path'] = osp.join(self.gt_folder,
+                                            data_info['gt_path'])
```
Comment (ckkelvinchan) [inline mmedit/datasets/sr_landmark_dataset.py:61]: A question, how do we generate the LQ images?

Comment (Yshuo-Li) [reply mmedit/datasets/sr_landmark_dataset.py:61]: Down-sampling

File: mmedit/datasets/sr_landmark_dataset.py (hunk 2 above)
Comment (Yshuo-Li) [reply mmedit/datasets/sr_landmark_dataset.py:52]: Sorry, don't need to be separated by a white space.

File: mmedit/datasets/sr_landmark_dataset.py (hunk 1 above)
Comment (Yshuo-Li) [reply mmedit/datasets/sr_landmark_dataset.py:14]: Faces only

Comment (ckkelvinchan) [reply mmedit/datasets/sr_landmark_dataset.py:14]: So would it be better to change it to `SRFacicalLandmarkDataset`? Also, the "general" in line 11 should be changed?

File: mmedit/datasets/sr_facical_landmark_dataset.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+        It loads the GT image path and landmark from the annotation file.
+        Each dict in the annotation file contains the image names, image
+        shape (usually for gt), bbox and landmark.
+
+        Returns:
+            dict: Returned dict for GT and landmark.
+                Contains: gt_path, bbox, shape, landmark.
+        """
+        data_infos = np.load(self.ann_file, allow_pickle=True)
```
Comment (innerlee) [inline mmedit/datasets/sr_facical_landmark_dataset.py:58]: Use `mmcv.load`. Benefit: it will support json automatically.

**Comment by innerlee** [comment]
A minor change

**Review by innerlee — APPROVED** [review]

File: mmedit/datasets/sr_facical_landmark_dataset.py (hunk 4 above)
Comment (Yshuo-Li) [reply mmedit/datasets/sr_facical_landmark_dataset.py:58]: But it doesn't support "npy" file, and here we don't need JSON data.

Comment (innerlee) [reply mmedit/datasets/sr_facical_landmark_dataset.py:58]: I see.

**Pull request merged**